pydriller/domain/commit.py

```diff
@@ -254,18 +255,62 @@ def methods(self) -> List[Method]:
         self._calculate_metrics()
         return self._function_list
 
-    def _calculate_metrics(self):
+    @property
+    def methods_before(self) -> List[Method]:
+        """
+        Return the list of methods in the file before the
+        change happened. Each method will have all specific
+        info, e.g. complexity, loc, name, etc.
+
+        :return: list of methods
+        """
+        self._calculate_metrics(include_before=True)
+        return self._function_list_before
+
+    @property
+    def changed_methods(self) -> List[Method]:
+        """
+        Return the list of methods that were changed. This analysis
+        is more complex because lizzard runs twice: for methods before
+        and after the change
+
+        :return: list of methods
+        """
+        new_methods = self.methods
+        old_methods = self.methods_before
+        added = self.diff_parsed['added']
+        deleted = self.diff_parsed['deleted']
+
+        methods_changed_new = set([
+            y for x in added for y in new_methods if x[0] <= y.end_line and x[0] >= y.start_line])
+        methods_changed_old = set([
+            y for x in deleted for y in old_methods if x[0] <= y.end_line and x[0] >= y.start_line
+        ])
+
+        return list(methods_changed_new.union(methods_changed_old))
```

Comment:
The union will probably fail for changed methods (but not for added or deleted). 
The reason is that for the same method, the object `Method` in `methods_changed_new` is different from the object `Method` in `methods_changed_old`. Hence we will have the same method twice.
This is just a supposition, we need more testing

Reply:
If this is the case, just consider one of the 2

Reply:
Yes, I was expecting this, but we have to decide which would be the Method's attribute to filter for. Is name suitable for detecting duplicates in this case?

Reply:
This is solved by adding  __hash__ and __eq__ to the 'Method' class. We should check the unique attributes there.